File: backend/internal/domain/admin_invitation.go
package domain

import "time"

type AdminInvitation struct {
	ID        uint64   `json:"id"`
	CompanyID uint64   `json:"companyId"`
	Email     string   `json:"email"`
	Role      RoleName `json:"role"`
	Name      string   `json:"name"`
	Status    string   `json:"status"`
	// Token はマジックリンク用の不透明 UUID（秘匿値なので json では返さない）。
	// 未設定値を NULL にして UNIQUE 制約に引っかけないため *string にしている。
	Token     *string   `json:"-"`
	ExpiresAt time.Time `json:"expiresAt"`
	CreatedAt time.Time `json:"createdAt"`
	// WorkspaceID は所属ワークスペースへの参照。CompanyID から dual-write されるため
	// 通常は必ず埋まっているが、Course と同じ理由で NULL を許容する型にする（FRESTYLE-297）。
	WorkspaceID *string `json:"workspaceId,omitempty"`
}

// WorkspaceRef は所属ワークスペースへの参照を返す。未設定(workspace_id = NULL)は NoWorkspace。
func (i AdminInvitation) WorkspaceRef() WorkspaceRef {
	if i.WorkspaceID == nil {
		return NoWorkspace()
	}
	return WorkspaceRefOf(*i.WorkspaceID)
}

const (
	InvitationStatusPending  = "pending"
	InvitationStatusAccepted = "accepted"
	InvitationStatusCanceled = "canceled"
)

File: backend-java/src/main/java/com/normanblog/frestyle/entity/AdminInvitation.java
package com.normanblog.frestyle.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 企業担当者を招待するためのマジックリンク招待。
 *
 * <p>初回ログイン時にこの行(pending)があれば、新規ユーザーの作成を許可し、role / company を反映する。
 */
@Entity
@Table(name = "invitations")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AdminInvitation {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(name = "company_id")
  private Long companyId;

  private String email;

  private String role;

  @Column(name = "display_name")
  private String displayName;

  private String status;

  // マジックリンク用の不透明トークン(秘匿値)。未設定を NULL にして UNIQUE 制約に当てないため null 許容。
  private String token;

  @Column(name = "expires_at")
  private Instant expiresAt;

  @Column(name = "created_at")
  private Instant createdAt;
}
